openmetadata-service/src/main/java/org/openmetadata/service/apps/AbstractNativeApplication.java

```diff
@@ -0,0 +1,76 @@
+package org.openmetadata.service.apps;
+
+import static org.openmetadata.service.apps.scheduler.AppScheduler.APP_INFO;
+import static org.openmetadata.service.apps.scheduler.AppScheduler.COLLECTION_DAO_KEY;
+
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.entity.app.AppSchedule;
+import org.openmetadata.schema.entity.app.AppType;
+import org.openmetadata.schema.entity.app.Application;
+import org.openmetadata.schema.entity.app.RuntimeContext;
+import org.openmetadata.service.apps.scheduler.AppScheduler;
+import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+
+@Slf4j
+public class AbstractNativeApplication implements NativeApplication {
+  private CollectionDAO collectionDAO;
+  private Application app;
+
+  @Override
+  public void init(Application app, CollectionDAO dao) {
+    this.collectionDAO = dao;
+    this.app = app;
+  }
+
+  @Override
+  public void triggerOnDemand(Object requestObj) {
+    // Validate Native Application
+    validateServerExecutableApp(app.getExecutionContext().getOnDemand());
+
+    // Trigger the application
+    AppScheduler.getInstance().triggerOnDemandApplication(app);
+  }
+
+  @Override
+  public void schedule(AppSchedule schedule) {
```

Comment:
where is this getting called from, can't we not send the App to schedule if its not native